test/cluster/mzcompose.py

```diff
@@ -3164,3 +3164,93 @@ def subscribe():
                     ), e
             for thread in threads:
                 assert not thread.is_alive(), f"Thread {thread.name} is still running"
+
+
+def workflow_test_refresh_mv_warmup_after_restart(
+    c: Composition, parser: WorkflowArgumentParser
+) -> None:
+    """
+    Test that bootstrapping selects an as_of for an MV dataflow in a way that allows it to warm up before its next
+    refresh.
+    """
+
+    with c.override(
+        Materialized(
+            additional_system_parameter_defaults={
+                "enable_refresh_every_mvs": "true",
+            },
+        ),
+        Testdrive(no_reset=True),
+    ):
+        c.down(destroy_volumes=True)
+
+        c.up("materialized")
+        c.up("clusterd1")
+
+        # Create a new cluster
+        c.sql("DROP CLUSTER IF EXISTS cluster1 CASCADE;")
+        c.sql(
+            "ALTER SYSTEM SET enable_unmanaged_cluster_replicas = true;",
+            port=6877,
+            user="mz_system",
+        )
+        c.sql(
+            """
+            CREATE CLUSTER cluster1 REPLICAS (replica1 (
+                STORAGECTL ADDRESSES ['clusterd1:2100'],
+                STORAGE ADDRESSES ['clusterd1:2103'],
+                COMPUTECTL ADDRESSES ['clusterd1:2101'],
+                COMPUTE ADDRESSES ['clusterd1:2102'],
+                WORKERS 1
+            ));
+            """
+        )
+
+        c.up("testdrive", persistent=True)
+
+        # Create an MV that will have its first refresh immediately, but it's next refresh is a long time away.
+        c.testdrive(
+            input=dedent(
+                """
+                > SET cluster = cluster1;
+                > CREATE TABLE t (a int);
+                > INSERT INTO t VALUES (100);
+                > CREATE MATERIALIZED VIEW mv WITH (REFRESH EVERY '1 day') AS
+                  SELECT count(*) FROM (SELECT generate_series(1,a) FROM t);
+
+                > SELECT * FROM mv;
+                100
+
+                > INSERT INTO t VALUES (1000);
+                """
+            )
+        )
+
+        # Restart environmentd to trigger the
+        c.kill("materialized")
+        c.up("materialized")
+
+        # Check that the dataflow hydrates, even though we are a long time away from the next refresh.
+        c.testdrive(
+            input=dedent(
+                """
+                > SET cluster = cluster1;
```

Comment:
That `SET cluster` isn't needed.

Reply:
Ah yes, I was being overly defensive here.